.github/workflows/main.yml

```diff
@@ -1,42 +1,196 @@
-name: Deploy GH Pages
-
-on:
-  push:
-    branches: 
-      - master 
-    paths:
-      - 'docs/*.md'
-      - 'docs/quickstart'
-      - '*.md'
-
-jobs:
-  deploy-gh-pages:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout master
-        uses: actions/checkout@v2
-        with:
-          path: master
-
-      - name: Checkout gh-pages branch
-        uses: actions/checkout@v2
-        with:
-          ref: 'gh-pages'
-          path: gh-pages
-    
-      - name: Copy site files
-        run: |
-          cp master/docs/*.md gh-pages/
-          cp -r master/docs/quickstart gh-pages/_pages
-          cp master/CODE_OF_CONDUCT.md master/CONTRIBUTING.md gh-pages/_pages/
-
-      - name: Commit GH Pages
-        run: |
-          cd gh-pages
-          git config user.name armada-admin
-          git config user.email admin@armadaproject.io
-          git add . _pages
-          git diff --quiet && git diff --staged --quiet || git commit -m "Updating Github Pages branch with latest Master changes"
-          git push
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+# Pre-merge TODO list
+# 1. change DOCKERTOKEN and DOCKERUSERNAME
+name: CI
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  GO111MODULE: "on"
+  GOPATH: "/home/runner/work/armada/armada/go"
+  GOCACHE: "/home/runner/.cache/go-build"
+
+on: 
+  push:
+    branches:
+      - '*'
+    pull_request:
+      branches:
+        - master
+    tags-ignore:
+      - '**'
+    paths-ignore:
+      - 'docs/*.md'
+      - 'docs/quickstart'
+      - '*.md'
+      - '**/release.yml'
+      - '**/pages.yml'
+
+jobs:
+  code_styles:
+    name: Code style
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+          
+      - name: Check TypeScript formatting
+        run: |
+          npm install && npm run fmt || exit 1
+          exit $(git status -s -uno | wc -l)
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada/internal/lookout/ui
+
+      - name: Check Go formatting
+        run: |
+          go install golang.org/x/tools/cmd/goimports@v0.1.1

```

Comment:
You do the goimports twice here.  (Check Go formatting and Check formatting are the same).

One should be ineffassign:
https://github.com/G-Research/armada/blob/master/.circleci/config.yml#L121

Also could we put them before Typescript for faster feedback

I also see your discussion on goimports vs gofmt. gofmt does not have the ability to enforce the import order we want in our project, or at least it didn't used to. If that has changed then we can use gofmt

Reply:
Updated jobs and removed sufficient scripts to update release summary  🚀 
here is [latest commit](https://github.com/G-Research/armada/pull/597/commits/11d2441b4de71e2804e41ef96d0ce76378090791)